pennylane/ops/op_math/adjoint_constructor.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule applies the symbolic operation that indicates the adjoint of an operator through the `adjoint` transform.
+"""
+from functools import wraps
+
+from pennylane.operation import Operator, AdjointUndefinedError
+from pennylane.queuing import QueuingContext, QueuingError
+from pennylane.tape import QuantumTape, stop_recording
+
+from .adjoint_class import Adjoint
+
+
+def _single_op_eager(op):
+    try:
+        return op.adjoint()
+    except AdjointUndefinedError:
+        return Adjoint(op)
+
+
+def _single_op_eager_update_queue(op):
```

Comment:
No urgency at all, but curious why this is not merged with above function with a kwarg for `update_queue`?

Reply:
Since recording on the surrounding context is halted when we turn the quantum function into a tape, `op` from a quantum function will not be in the queue already. We don't need to update its metadata.

I'm turning this distinction into a keyword argument for a single private function `_single_op_eager`.